src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,13 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null) ? true : _property.IsExpressionBodied;
```

Comment:
This logic doesn't feel right. I think accessor's body should always take a priority, if present.